src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -749,6 +749,56 @@ private static String captureSystemStream(Runnable runnable, PrintStream stream,
         return out.toString();
     }
 
+    @CommandLineProgramProperties(
+            summary = "tool with boundaries",
+            oneLineSummary = "tools with boundaries",
+            programGroup = TestProgramGroup.class
+    )
+    public class WithBoundariesArguments {
+
+        @Argument(doc = "Integer in the range [0, 30]", optional = true, minValue = 0, maxValue = 30)
+        public int integerArg = 20;
+
+        @Argument(doc = "String argument", optional = true, minValue = 0, maxValue = 30)
```

Comment:
Per my comment above, I think this case should throw, since the range is not honored/enforced. We should keep a test for this though to make sure it fails as we think it should.